### PR TITLE
Destroy secrets when we're done with them

### PR DESCRIPTION
**Pull request by grbean**
Signed-off-by: Greg Bean <greg.bean@intel.com>

### CODE REVIEW

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java [hunk 1]
```diff
@@ -1154,38 +1165,56 @@ protected KeyExchangeResult getEcdhSharedSecret(byte[] message, Composite ownSta
...
       final KeyFactory factory = getKeyFactoryInstance(Const.EC_ALG_NAME);
+      final KeyAgreement keyAgreement = getKeyAgreementInstance(Const.ECDH_ALG_NAME);
       final ECPrivateKey ownKey = (ECPrivateKey) factory.generatePrivate(privateSpec);
+      final byte[] sharedSecret;
+      try {
 
-      final ECPoint w = new ECPoint(new BigInteger(1, theirX), new BigInteger(1, theirY));
-      final ECPublicKeySpec publicSpec = new ECPublicKeySpec(w, ownKey.getParams());
+        final ECPoint w = new ECPoint(new BigInteger(1, theirX), new BigInteger(1, theirY));
+        final ECPublicKeySpec publicSpec = new ECPublicKeySpec(w, ownKey.getParams());
```
Comment (johneasterday) [inline protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java:1174]: is this the correct indentation? checkstyle accepts this?

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java [hunk 2]
```diff
@@ -1154,38 +1165,56 @@ protected KeyExchangeResult getEcdhSharedSecret(byte[] message, Composite ownSta
...
 
-      final PublicKey theirKey = factory.generatePublic(publicSpec);
+        final PublicKey theirKey = factory.generatePublic(publicSpec);
```
Comment (johneasterday) [inline protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java:1176]: indentation correct?

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java (hunk 1 above)
Comment (johneasterday) [reply protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java:1174]: never mind, I see it now

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java (hunk 2 above)
Comment (johneasterday) [reply protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java:1176]: never mind, I see it now

**Review by johneasterday — APPROVED** [review]

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java (hunk 2 above)
Comment (grbean) [reply protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java:1176]: Yes.  Code's been moved inside of a try{} block and so get moved down one layer.

File: protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java (hunk 1 above)
Comment (grbean) [reply protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java:1174]: Yes.  Code's been moved inside of a try{} block and so get moved down one layer.

**Review by trbehera — APPROVED** [review]

**Pull request merged**